lib/dependencies.sh

```diff
@@ -305,3 +316,47 @@ npm_prune_devdependencies() {
     meta_set "skipped-prune" "false"
   fi
 }
+
+pnpm_install() {
+  local build_dir=${1:-}
+
+  echo "Running 'pnpm install' with pnpm-lock.yaml"
+  cd "$build_dir" || return
+
+  monitor "pnpm-install" pnpm install --prod=false --frozen-lockfile 2>&1
+
+  # prune the store after install to remove any package versions which may have been upgraded/removed
+  # since the last install.
+  suppress_output pnpm store prune
```

Comment:
I think it makes sense to do this a little less frequently than every build. If you think about review apps, application rollbacks, apps deployed on different branches, there are bound to be some dependencies that get removed, but used again on a later build. We don't need to keep everything they've ever shipped, but I think it's nice to keep a backlog of say 20 builds in here.

Reply:
Looks like we've got this set to `40` in the CNB: https://github.com/heroku/buildpacks-nodejs/blob/main/buildpacks/nodejs-pnpm-install/src/store.rs#L1. Maybe we should aim for something similar here?

Reply:
👍 i'll make that change

Reply:
Added in #1231